onnxruntime/core/providers/qnn/builder/opbuilder/simple_op_builder.cc

```diff
@@ -261,10 +267,118 @@ Status SimpleOpBuilder::ProcessAttributesAndOutputs(QnnModelWrapper& qnn_model_w
+ * specific values. Returns true if the quantization parameters were overridden.
+ *
+ * \param op_type The ONNX operator type.
+ * \param qnn_data_type The QNN tensor data type.
+ * \param quant_params Output scale/offset parameter that may be overridden.
+ * \return True if the offset and scale were overridden.
+ */
+static bool OverrideQuantParams(const std::string& op_type, Qnn_DataType_t qnn_data_type,
+                                Qnn_ScaleOffset_t& quant_params) {
+  const int32_t orig_offset = quant_params.offset;
+  const float orig_scale = quant_params.scale;
+
+  if (op_type == "Sigmoid") {
+    switch (qnn_data_type) {
+      case QNN_DATATYPE_UFIXED_POINT_16:
+        quant_params.offset = 0;
+        quant_params.scale = 1.0f / 65536.0f;
+        break;
+      case QNN_DATATYPE_SFIXED_POINT_16:
+        quant_params.offset = 0;
+        quant_params.scale = 1.0f / 32768.0f;
+        break;
+      default:
+        break;  // Do nothing.
+    }
+  }
+
+  if (op_type == "Tanh") {
+    switch (qnn_data_type) {
+      case QNN_DATATYPE_UFIXED_POINT_16:
+        quant_params.offset = -32768;
+        quant_params.scale = 1.0f / 32768.0f;
+        break;
+      case QNN_DATATYPE_SFIXED_POINT_16:
+        quant_params.offset = 0;
+        quant_params.scale = 1.0f / 32768.0f;
+        break;
+      default:
+        break;  // Do nothing.
+    }
+  }
+
+  return quant_params.offset != orig_offset || quant_params.scale != orig_scale;
+}
+
+/**
+ * Processes the output for Sigmoid or Tanh operators and creates the corresponding QNN operator.
+ * These operator types are handled separately because QNN requires 16-bit QDQ Sigmoid and Tanh operators to use
+ * specific scale and zero-point values regardless of floating-point range.
+ *
+ * \param qnn_model_wrapper The QNN model wrapper object.
+ * \param node_unit The QDQ node unit for the Sigmoid or Tanh node.
+ * \param input_names List of input names.
+ * \param param_tensor_names List of param tensor names.
+ * \param logger Logger used to report information.
+ * \param do_op_validation True if the new QNN node should be validated.
+ */
+Status SimpleOpBuilder::ProcessSigmoidOrTanhOutput(QnnModelWrapper& qnn_model_wrapper,
+                                                   const NodeUnit& node_unit,
+                                                   std::vector<std::string>&& input_names,
+                                                   std::vector<std::string>&& param_tensor_names,
+                                                   const logging::Logger& logger,
+                                                   bool do_op_validation) const {
+  const std::string& op_type = node_unit.OpType();
+  const auto& output = node_unit.Outputs()[0];
+  const std::string& output_name = output.node_arg.Name();
+
+  OnnxInputInfo output_info = {};
+  ORT_RETURN_IF_ERROR(qnn_model_wrapper.GetOnnxInputInfo(output, output_info));
```

Comment:
>GetOnnxInputInfo [](start = 40, length = 16)

Could you leave a TODO comment for QnnModelWrapper::GetOnnxInputInfo, we need to rename it to make it meaningful.

Reply:
Yep, added a todo.